tests/Functional/Connection/FetchTest.php

```diff
@@ -176,7 +176,7 @@ public function testIterateKeyValueOneColumn(): void
             ->getDummySelectSQL();
 
         $this->expectException(NoKeyValue::class);
-        iterator_to_array($this->connection->iterateKeyValue($sql));
+        $assign = iterator_to_array($this->connection->iterateKeyValue($sql));
```

Comment:
I think at least Psalm might be complaining about this at some point in time or when using a stricter level. [There is even a blog post about that](https://psalm.dev/articles/better-unused-variable-detection)
I had the same issue on a paid project with my colleague @dFayet , and resorted to an ignore rule on the `tests` directory. @morozov , what's your opinion on this?

Reply:
A proper way to fix this might be to rework the usage of `iterator_to_array()` to a set of lower-level iterator functions like `reset()`, `next()` to trigger the fetch. Not sure how PHPStan decides whether the return value of a given function should be used or not.

Alternatively, this could be temporarily suppressed in the configuration.

Reply:
I tried to change it

Reply:
ok the error disappeared. any idea why?

Reply:
What error?